include/algorithms/public/SineFeatureExtraction.hpp

```diff
@@ -0,0 +1,98 @@
+/*
+Part of the Fluid Corpus Manipulation Project (http://www.flucoma.org/)
+Copyright 2017-2019 University of Huddersfield.
+Licensed under the BSD-3 License.
+See license.md file in the project root for full license information.
+This project has received funding from the European Research Council (ERC)
+under the European Union’s Horizon 2020 research and innovation programme
+(grant agreement No 725899).
+*/
+
+#pragma once
+
+#include "WindowFuncs.hpp"
+#include "../util/AlgorithmUtils.hpp"
+#include "../util/FFT.hpp"
+#include "../util/FluidEigenMappings.hpp"
+#include "../util/PartialTracking.hpp"
+#include "../util/PeakDetection.hpp"
+#include "../../data/FluidIndex.hpp"
+#include "../../data/FluidMemory.hpp"
+#include "../../data/TensorTypes.hpp"
+#include <Eigen/Core>
+#include <cmath>
+#include <queue>
+
+namespace fluid {
+namespace algorithm {
+
+
+class SineFeatureExtraction 
+{
+  using ArrayXd = Eigen::ArrayXd;
+  using VectorXd = Eigen::VectorXd;
+  using ArrayXcd = Eigen::ArrayXcd;
+  template <typename T>
+  using vector = rt::vector<T>;
+
+public:
+  SineFeatureExtraction(Allocator& alloc)
```

Comment:
```suggestion
SineFeature(Allocator& alloc)
```

Reply:
We will, at some point, want to pass maximal window / fftSizes in by the constructor too

Reply:
for now, done.